src/HealthChecks/Abstractions/src/HealthCheckRegistration.cs

```diff
@@ -35,7 +35,8 @@ public sealed class HealthCheckRegistration
         /// is <c>null</c>, then <see cref="HealthStatus.Unhealthy"/> will be reported.
         /// </param>
         /// <param name="tags">A list of tags that can be used for filtering health checks.</param>
-        public HealthCheckRegistration(string name, IHealthCheck instance, HealthStatus? failureStatus, IEnumerable<string> tags)
+        /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+        public HealthCheckRegistration(string name, IHealthCheck instance, HealthStatus? failureStatus, IEnumerable<string> tags, TimeSpan? timeout = null)
```

Comment:
So, this is a breaking change, we can't do this 😆 It is a binary breaking change to add a new parameter with a default value. So even though people's code would *compile* it will break existing compiled code.

Reply:
So you'll need to add another overload of each of these.

Reply:
Is running assemblies compiled against 2.x with a runtime 3.x supported ? I thought dotnetcore rollforward rules applied (cf. https://docs.microsoft.com/en-us/dotnet/core/versions/selection#framework-dependent-apps-roll-forward )

Reply:
Ok forget my previous message, I totally missed that it can be executed in other contexts than aspnetcore 😅

Reply:
Yes, and libraries are a thing too. Most libraries are pretty naive about what version of our stuff the target. So they need to be compatible with a range of things.

We really intend for an area like health checks to be super stable.